electrumx/server/session.py

```diff
@@ -1245,11 +1247,37 @@ async def estimatefee(self, number, mode=None):
         mode: CONSERVATIVE or ECONOMICAL estimation mode
         '''
         number = non_negative_integer(number)
+        # use whitelist for mode, otherwise it would be easy to force a cache miss:
+        if mode not in self.coin.ESTIMATEFEE_MODES:
+            raise RPCError(BAD_REQUEST, f'unknown estimatefee mode: {mode}')
         self.bump_cost(2.0)
```

Comment:
Maybe cache hit should have a lower cost, since miss is penalized further down?

Reply:
done